src/snowcli/cli/registry/commands.py

```diff
@@ -14,10 +16,108 @@
 
 
 @app.command("token")
-@with_output
 @global_options_with_connection
-def token(environment: str = ConnectionOption, **options):
+def token(**options):
     """
     Gets the token from environment to use for authenticating with the registry.
     """
-    return ObjectResult(get_token(environment))
+    sys.stdout.write(json.dumps(RegistryManager().get_token()))
+
+
+@app.command("list-images")
+@global_options_with_connection
+def list_images(
+    repo_name: str = typer.Option(
+        ...,
+        "--repository_name",
+        "-r",
+        help="Name of the image repository as seen in `show image repositories`",
+    ),
+    **options,
+):
+    database = RegistryManager().get_database()
+    schema = RegistryManager().get_schema()
+    url = RegistryManager().get_repository_url(repo_name)
+    api_url = RegistryManager().get_repository_api_url(url)
+    bearer_login = RegistryManager().login_to_registry(api_url)
+
+    repos = []
+    query: Optional[str] = f"{api_url}/_catalog?n=10"
+
+    while query is not None:
+        # Make paginated catalog requests
+        response = requests.get(
+            query, headers={"Authorization": f"Bearer {bearer_login}"}
+        )
+
+        if response.status_code != 200:
+            print("Call to the registry failed", response.text)
```

Comment:
```suggestion
            return MessageResult("Call to the registry failed {response.text}")
```
However, I would consider throwing an exception